### PR TITLE
Initial framework for SPLIT_MODULE option

### PR DESCRIPTION
**Pull request by sbc100**
This option uses the new wasm-split tool in binaryen to generate profile
data and then split a module into a primary and secondary component
modules that can be linked together at runtime.

Using this option will current generate an `experimental` warning to let
folks know its a WIP.

### CODE REVIEW

**Comment by sbc100** [comment]
@tlively @awtcode

**Comment by awtcode** [comment]
Txs for doing this @sbc100 :) Really appreciate it!

File: src/settings.js [hunk 1]
```diff
@@ -1647,6 +1647,15 @@ var PURE_WASI = 0;
 // - WASM2JS (WASM=0)
 var IMPORTED_MEMORY = 0;
 
+// Generate code to loading split wasm modules.
+// This option will autoamtically generate two wasm files as output, one
+// with the `.orig` suffix and one without.  The default file (without
+// the suffix) when run will generate instrumation data can later be
```
Comment (tlively) [inline src/settings.js:1653]:
```suggestion
// the suffix) when run will generate instrumentation data that can later be
```

**Comment by sbc100** [comment]
@tlively I think we can land this initial verion which is essentially just laying the ground work.   Then any of us to pick it up and continue to the work.   Or maybe it not enough functionality to land on its own?

**Review by tlively — APPROVED** [review]
Yes, this looks good to me!

**Pull request merged**